### ISSUE TITLE
Potentially imprecise description of reportUnnecessaryIsInstance

### ISSUE THREAD

**Issue by kkom** [microsoft/pyright#11111]
**Describe the bug**

Pyright's [diagnostic rules documentation](https://microsoft.github.io/pyright/#/configuration?id=type-check-diagnostics-settings) states this:

> **reportUnnecessaryIsInstance** [boolean or string, optional]: Generate or suppress diagnostics for `isinstance` or `issubclass` calls where the result is statically determined to be always true or always false. Such calls are often indicative of a programming error. The default value for this setting is `"none"`.

However, I think that Pyright will never determine the result to be "always false" – as it always allows for the possibility that a variable subclasses multiple types. 

Unless I'm missing some scenario, I think the docs should say only "where the result is statically determined to be always true".

**Code or Screenshots**

No diagnostics generated in the below code snippet:

```python
from typing import reveal_type

x = 1

if isinstance(x, str):
    reveal_type(x)  # Type of "x" is "<subclass of Literal[1] and str>"

c = int

if issubclass(c, str):
    reveal_type(c)  # Type of "c" is "type[<subclass of int and str>]"
```

**VS Code extension or command-line**

VS Code extension, Pylance 2025.9.1

**Comment by erictraut** [comment]
The documentation is correct. Pyright will detect an "always false" condition in cases that involve `@final` classes.

```python
# pyright: strict

from typing import final

@final
class A: ...

class B: ...

def test(x: A):
    if isinstance(x, B):  # reportUnnecessaryIsinstance: A is never an instance of B
        ...
```

**Comment by kkom** [comment]
Thank you for this example! Feel free to close the issue.
